app/ui/browser/model/index.js

```diff
@@ -29,6 +34,8 @@ export const State = Immutable.Record({
   // This is a list of Page objects (see below)
   pages: Immutable.List(),
 
+  pageIds: Immutable.Set(),
```

Comment:
`pages.map` creates a new object, so you don't get the referential transparency you need to do a quick check in `shouldComponentUpdate`.